Applications/Spire/Source/Common/LocalRangeInputModel.cpp

```diff
@@ -0,0 +1,46 @@
+#include "Spire/Spire/LocalRangeInputModel.hpp"
+#include "Spire/Spire/Utility.hpp"
+
+using namespace Nexus;
+using namespace Spire;
+
+LocalRangeInputModel::LocalRangeInputModel(
+    const std::vector<ChartValue> data)
+    : m_data(data),
+      m_minimum_value(0),
+      m_maximum_value(std::numeric_limits<int>::max()) {
```

Comment:
These initializations are incorrect.

Reply:
What should the defaults be if there is no data?

Reply:
If there is no data then the min should equal the max. Also avoid situations where you initialize data and then overwrite it, for example:

    auto x = 0;
    if(some_condition) {
      x = 10;
    }

Instead prefer:

    if(some_condition) {
      x = 10;
    } else {
      x = 0;
    }

In other words, remove the initialization from 10 and 11 and initialize those in the body of the constructor.